### PR TITLE
known_audio_plugins

### PR DESCRIPTION
**Pull request by RomanPudashkin**

### CODE REVIEW

File: src/framework/audio/CMakeLists.txt [hunk 1]
```diff
@@ -238,6 +252,10 @@ elseif (OS_IS_LIN)
     set(MODULE_LINK ${MODULE_LINK} ${ALSA_LIBRARIES} pthread )
 endif()
 
+if (MUE_BUILD_UNIT_TESTS)
+    add_subdirectory(tests)
+endif()
```
Comment (igorkorsukov) [inline src/framework/audio/CMakeLists.txt:257]: please, move to end of file (to be the same as everywhere)

File: src/framework/audio/audiomodule.cpp [hunk 2]
```diff
@@ -66,6 +70,8 @@ static std::shared_ptr<Playback> s_playbackFacade = std::make_shared<Playback>()
 
 static std::shared_ptr<SoundFontRepository> s_soundFontRepository = std::make_shared<SoundFontRepository>();
 
+static std::shared_ptr<KnownAudioPluginsRegister> s_knownAudioPluginsRegister = std::make_shared<KnownAudioPluginsRegister>();
```
Comment (igorkorsukov) [inline src/framework/audio/audiomodule.cpp:73]: Please prefer to create objects in the registerExports method

File: src/framework/audio/internal/plugins/knownaudiopluginsregister.cpp [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    }
+
+    for (const io::path_t& infoPath : paths.val) {
+        RetVal<ByteArray> file = fileSystem()->readFile(infoPath);
+        if (!file.ret) {
+            LOGE() << file.ret.toString();
+            continue;
+        }
+
+        QJsonDocument json = QJsonDocument::fromJson(file.val.toQByteArrayNoCopy());
```
Comment (igorkorsukov) [inline src/framework/audio/internal/plugins/knownaudiopluginsregister.cpp:115]: Please use our JsonDocument (not Qt)

File: src/framework/audio/internal/plugins/knownaudiopluginsregister.cpp [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    obj.insert(QStringLiteral("meta"), metaToJson(info.meta));
+    obj.insert(QStringLiteral("path"), info.path.toQString());
+    obj.insert(QStringLiteral("enabled"), info.enabled);
+
+    if (info.errorCode != 0) {
+        obj.insert(QStringLiteral("errorCode"), info.errorCode);
+    }
+
+    io::path_t path = pluginInfoPath(info.meta.id);
+    Ret ret = fileSystem()->writeFile(path, ByteArray::fromQByteArrayNoCopy(QJsonDocument(obj).toJson()));
```
Comment (igorkorsukov) [inline src/framework/audio/internal/plugins/knownaudiopluginsregister.cpp:183]: It is dangerous to call `fromQByteArrayNoCopy` for a temporary object

File: src/framework/audio/tests/knownaudiopluginsregistertest.cpp [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+        .WillByDefault(Return(m_knownAudioPluginsDir));
+    }
+
+    ByteArray pluginInfoToJson(const AudioPluginInfo& info) const
+    {
+        const std::map<AudioResourceType, QString> RESOURCE_TYPE_TO_STR {
+            { AudioResourceType::VstPlugin, "VstPlugin" },
+        };
+
+        QJsonObject attributesObj;
```
Comment (igorkorsukov) [inline src/framework/audio/tests/knownaudiopluginsregistertest.cpp:62]: please use our Json (not Qt)

**Comment by RomanPudashkin** [comment]
@igorkorsukov fixed, thanks!

File: src/framework/audio/audiomodule.cpp [hunk 6]
```diff
@@ -107,6 +113,16 @@ std::string AudioModule::moduleName() const
...
 {
+    s_audioConfiguration = std::make_shared<AudioConfiguration>();
+    s_audioWorker = std::make_shared<AudioThread>();
+    s_audioBuffer = std::make_shared<AudioBuffer>();
+    s_audioOutputController = std::make_shared<AudioOutputDeviceController>();
+    s_fxResolver = std::make_shared<FxResolver>();
+    s_synthResolver = std::make_shared<SynthResolver>();
+    s_playbackFacade = std::make_shared<Playback>();
+    s_soundFontRepository = std::make_shared<SoundFontRepository>();
+    s_knownAudioPluginsRegister = std::make_shared<KnownAudioPluginsRegister>();
```
Comment (igorkorsukov) [inline src/framework/audio/audiomodule.cpp:124]: Excellent!

**Review by igorkorsukov — APPROVED** [review]

**Pull request merged**